junebug/tests/test_api.py

```diff
@@ -57,14 +57,26 @@ def test_http_error(self):
 
     @inlineCallbacks
     def test_get_channel_list(self):
+        redis = yield self.get_redis()
+        config = self.default_channel_config
+
         resp = yield self.get('/channels')
-        yield self.assert_response(
-            resp, http.INTERNAL_SERVER_ERROR, 'generic error', {
-                'errors': [{
-                    'message': '',
-                    'type': 'NotImplementedError',
-                    }]
-                })
+        yield self.assert_response(resp, http.OK, 'channels listed', [])
+
+        yield Channel(redis, {}, config, u'test-channel-1').save()
+
+        resp = yield self.get('/channels')
+        yield self.assert_response(resp, http.OK, 'channels listed', [
+            u'test-channel-1',
```

Comment:
To do this it would be an extra redis request per channel, which could be quite costly. On the other hand, if for most use cases we're going to have to get all the data for all the channels, then it's better to send it all on one request. How do we see this being used?